finality-aleph/src/data_io/data_store.rs

```diff
@@ -105,6 +107,43 @@ impl Default for DataStoreConfig {
     }
 }
 
+#[derive(Debug)]
+pub enum DataStoreError {
+    BlockImportStreamClosed,
+    FinalizedBlocksStreamClosed,
+    NetworkMessagesTerminated,
+}
+
+impl Display for DataStoreError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            DataStoreError::BlockImportStreamClosed => {
+                write!(f, "Block import notification stream was closed.")
+            }
+            DataStoreError::FinalizedBlocksStreamClosed => {
+                write!(f, "Finalized block import notification stream was closed.")
+            }
+            DataStoreError::NetworkMessagesTerminated => {
+                write!(f, "Stream with network messages was closed.")
+            }
+        }
+    }
+}
+
+impl DataStoreError {
```

Comment:
Same comments as in with the other error – name could just be `Error` and constructors are not necessary.